### ISSUE TITLE
Threads are created anew for each MT rendering operation

### ISSUE THREAD

**Issue by StrayLightning** [StrayLightning/aeolius#24]
The current MT harness creates new worker threads for each MT job, which is incredibly inefficient.  This should be considered the highest priority performance issue.

**Comment by StrayLightning** [comment]
After thinking about this for a while, here are my current thoughts:

Having a thread pool for a server makes a lot of sense.  However, having thread pools for workers that will go off and carry out many different types of rendering operation does not fit so well.  In general, attempting to use thread pools will vastly complicate the code for currently indeterminate performance benefit.

Rayon works nicely for divide and conquer approaches.  It produces very succinct parellelised code.   However, it does mean that the application completely defers to Rayon to decide when to actually spawn another thread based on the dynamic load, and gives up on reuse of threads.  Using a recursive method to split the patches to be rendered works well, and with the Vulkan renderer, the code is extremely pleasing to the eye.  However, with the GL renderer the necessity of transferring the data to the master thread for flushing makes re-architecting the code a bit of a nightmare.

Rayon's parallel iterators are not such a good fit for Wyvern/Aeolius as wrangling the Renderer and ThreadData references to appease the borrow checker becomes a nightmare.  They work better with functions than with methods.

TL;DR: Moving to thread reuse should be an experimental task to investigate the potential performance benefit.  Doing so would make the code more complex and therefore less maintainable.  In future, if the Vulkan renderer became the sole target, a switch to Rayon or similar could result in very terse code.

Closing this for now.
